src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    public void indexValidator(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not valid");
+        }
+    }
+
+    public Object[] grow(T[] elementData, int size) {
+        if ((size + 1) >= elementData.length) {
+            int newCapacity = elementData.length + (elementData.length >> 1);
+            T[] newElementData = (T[]) new Object[newCapacity];
+            System.arraycopy(elementData, 0, newElementData, 0, size);
+            elementData = newElementData;
+        }
+        return elementData;
+    }
+
     @Override
     public void add(T value) {
-
+        elementData = (T[]) grow(elementData, size);
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not valid");
+        }
```

Comment:
Potentially repeating code should be extracted into private method.
```suggestion
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException("Index is not valid");
        }
```